packages/beeq/src/components/tag/bq-tag.tsx

```diff
@@ -0,0 +1,233 @@
+
+  /** The default type of the tag component */
+  @Prop({ reflect: true }) type: TTagType = 'default';
+
+  /** The variant of tag to apply on top of the variant */
+  @Prop({ reflect: true }) variant: TTagVariant = 'default';
+
+  /** If true, the button will be disabled (no interaction allowed) */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /** If true, the option is selected and active */
+  @Prop({ reflect: true }) selected: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('size')
+  @Watch('type')
+  @Watch('variant')
+  checkPropValues() {
+    validatePropValue(TAG_SIZE, 'medium', this.el, 'size');
+    validatePropValue(TAG_TYPE, 'default', this.el, 'type');
+    validatePropValue(TAG_VARIANT, 'default', this.el, 'variant');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Callback handler to be called when the tag is removable  */
+  @Event() bqHide: EventEmitter;
+
+  /** Callback handler to be called when the tag is not removable */
+  @Event() bqShow: EventEmitter;
+
+  /** Handler to be called when tag is clicked */
+  @Event() bqClick: EventEmitter<HTMLBqTagElement>;
+
+  /** Handler to be called when tag is focused */
+  @Event() bqFocus: EventEmitter<HTMLBqTagElement>;
+
+  /** Handler to be called when the tag key is pressed */
+  @Event() bqKeyDown: EventEmitter<KeyboardEvent>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /** Method to be called to remove the tag component */
+  @Method()
+  async hide(): Promise<void> {
+    this.handleHide();
+  }
+
+  /** Method to be called to show the alert component */
+  @Method()
+  async show(): Promise<void> {
+    this.handleShow();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleHide = () => {
+    const ev = this.bqHide.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = false;
+    }
+  };
+
+  private handleShow = () => {
+    const ev = this.bqShow.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+    }
+  };
+
+  private handleClick = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    const ev = this.bqClick.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.selected = !this.selected;
+    }
+  };
```

Comment:
I think that if the tag is `removable` we should avoid triggering the click event handler.

```suggestion
  private handleClick = (event: Event) => {
    if (this.removable) return;
    
    const ev = this.bqClick.emit(this.el);
    if (!ev.defaultPrevented) {
      this.selected = !this.selected;
    }
  };
```